workshops/forms.py

```diff
@@ -1094,3 +1206,283 @@ class Meta:
+            'notes': TextInput,
+        }
+
+
+class BulkDiscardProgressesForm(forms.Form):
+    """Form used to bulk discard all TrainingProgresses associated with
+    selected trainees."""
+
+    trainees = forms.ModelMultipleChoiceField(queryset=Person.objects.all())
+
+    helper = BootstrapHelper(add_submit_button=False,
+                             form_tag=False,
+                             display_labels=False)
+
+    SUBMIT_POPOVER = '''<p>Discarded progress will be displayed in the following
+    way: <span class='label label-default'><strike> Discarded
+    </strike></span>.</p>
+
+    <p>If you want to permanently remove records from system,
+    click one of the progress labels and, then, click "delete" button.</p>'''
+
+    helper.layout = Layout(
+        # no 'trainees' -- you should take care of generating it manually in
+        # the template where this form is used
+
+        # We use formnovalidate on submit button to disable browser
+        # validation. This is necessary because this form is used along with
+        # BulkAddTrainingProgressForm, which have required fields. Both forms
+        # live inside the same <form> tag. Without this attribute, when you
+        # click the following submit button, the browser reports missing
+        # values in required fields in BulkAddTrainingProgressForm.
+        FormActions(
+            Submit('discard',
+                   'Discard all progress of selected trainees',
+                   formnovalidate='formnovalidate',
+                   **{
+                       'data-toggle': 'popover',
+                       'data-html': 'true',
+                       'data-content': SUBMIT_POPOVER,
+                   }),
+            HTML('<a bulk-email-on-click class="btn btn-primary">'
+                 'Mail selected trainees</a>&nbsp;'),
+        )
+    )
+
+    class Meta:
+        model = TrainingProgress
+
+
+class BulkChangeTrainingRequestForm(forms.Form):
+    """Form used to bulk discard training requests or bulk unmatch trainees
+    from trainings."""
+
+    requests = forms.ModelMultipleChoiceField(
+        queryset=TrainingRequest.objects.all())
+
+    helper = BootstrapHelper(add_submit_button=False,
+                             form_tag=False,
+                             display_labels=False)
+    helper.layout = Layout(
+        # no 'requests' -- you should take care of generating it manually in
+        # the template where this form is used
+
+        # We use formnovalidate on submit buttons to disable browser
+        # validation. This is necessary because this form is used along with
+        # BulkMatchTrainingRequestForm, which have required fields. Both
+        # forms live inside the same <form> tag. Without this attribute,
+        # when you click one of the following submit buttons, the browser
+        # reports missing values in required fields in
+        # BulkMatchTrainingRequestForm.
+        FormActions(
```

Comment:
That's right, `.form-actions` class was removed in v3, but `FormActions` does its job: intent buttons as expected.